readthedocs/doc_builder/python_environments.py

```diff
@@ -255,21 +273,14 @@ def install_core_requirements(self):
                 'readthedocs-sphinx-ext<0.6'
             ])
 
-        cmd = [
-            'python',
-            self.venv_bin(filename='pip'),
-            'install',
-            '--upgrade',
-            '--cache-dir',
-            self.project.pip_cache_path,
-        ]
+        cmd = pip_install_cmd
```

Comment:
This is tricky. Here you are not copying the list but creating a new pointer to it.

I checked the code and the `pip_install_cmd` is not used again in this function, but in that case it would be garbage.

I suppose what you want to do here is `copy.copy(pip_install_cmd)`.

Reply:
I'm creating a new reference below when concatenating the list, but explicit is better.

Reply:
```
>>> pip_install_cmd = [1, 2, 3]
>>> cmd = pip_install_cmd
>>> cmd += [4, 5, 6]
>>> cmd
[1, 2, 3, 4, 5, 6]
>>> pip_install_cmd
[1, 2, 3, 4, 5, 6]
>>> 
```

That's where I mean the problem is.

Reply:
Yup, but that only happens when using `+=`, if you do `cmd = cmd + [4, 5, 6]`, it creates a new reference p: weird python